server/imports/core/connection/index.js

```diff
@@ -279,105 +431,318 @@ Connection.prototype = {
+        const rxSslCAFile = /.*sslCAFile=(.+\.pem).*/;
+        const matchSslCAFile = connectionUrl.match(rxSslCAFile);
+        mongoShellParams.push('--sslCAFile');
+        mongoShellParams.push(matchSslCAFile[1]);
+      }
+      if (connectionUrl.includes('sslPEMKeyFile=')) {
+        const rxSslPemKeyFile = /.*sslPEMKeyFile=(.+\.pem).*/;
+        const matchSslPemKeyFile = connectionUrl.match(rxSslPemKeyFile);
+        mongoShellParams.push('--sslPEMKeyFile');
+        mongoShellParams.push(matchSslPemKeyFile[1]);
+      }
+      if (connectionUrl.includes('sslPEMKeyFilePassword=')) {
+        const rxSslPemKeyFilePass = /.*sslPEMKeyFilePassword=(.+)&.*/;
+        const matchSslPemKeyFilePass = connectionUrl.match(rxSslPemKeyFilePass);
+        mongoShellParams.push('--sslCertificateKeyFilePassword');
+        mongoShellParams.push(matchSslPemKeyFilePass[1]);
+      }
+      if (connectionUrl.includes('sslCertificateSelector=')) {
+        const rxSslCertificateSelector = /.*sslCertificateSelector=(.+)&.*/;
+        const matchSslCertificateSelector = connectionUrl.match(
+          rxSslCertificateSelector
+        );
+        mongoShellParams.push('--sslCertificateSelector');
+        mongoShellParams.push(matchSslCertificateSelector[1]);
+      }
+    }
+    return mongoShellParams;
+  },
+
   getConnectionOptions(connection) {
     const result = { useNewUrlParser: true, useUnifiedTopology: true };
-    if (connection.authenticationType === 'mongodb_x509') addSSLOptions(connection.mongodb_x509, result);
-    if (connection.ssl && connection.ssl.enabled) addSSLOptions(connection.ssl, result);
-    if (connection.options && connection.options.connectWithNoPrimary) result.connectWithNoPrimary = true;
+    if (connection.authenticationType === 'mongodb_x509') {
+      addSSLOptions(connection.mongodb_x509, result);
+    }
+    if (connection.ssl && connection.ssl.enabled) {
+      addSSLOptions(connection.ssl, result);
+    }
+    if (connection.options && connection.options.connectWithNoPrimary) {
+      result.connectWithNoPrimary = true;
+    }
 
     // added authSource to here to provide same authSource as DB name if it's not provided when connection is being used by URL
-    if (connection.authenticationType === 'mongodb_cr' || connection.authenticationType === 'scram_sha_1' || connection.authenticationType === 'scram_sha_256') {
-      if (connection[connection.authenticationType].authSource) result.authSource = connection[connection.authenticationType].authSource;
-      else result.authSource = connection.databaseName;
-    } else if (connection.authenticationType === 'gssapi' || connection.authenticationType === 'plain') {
+    if (
+      connection.authenticationType === 'mongodb_cr' ||
+      connection.authenticationType === 'scram_sha_1' ||
+      connection.authenticationType === 'scram_sha_256'
+    ) {
+      if (connection[connection.authenticationType].authSource) {
+        result.authSource =
```

Comment:
Codacy has a fix for the issue: [There should be no line break before or after '='.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
        result.authSource =          connection[connection.authenticationType].authSource;
```